spring-kotlin/aws-ssh-server-manager/src/main/kotlin/dev/santos/awssshservermanager/mapper/UserMapper.kt

```diff
@@ -0,0 +1,42 @@
+package dev.santos.awssshservermanager.mapper
+
+import dev.santos.awssshservermanager.dto.CreateUserDto
+import dev.santos.awssshservermanager.model.User
+import org.mapstruct.InjectionStrategy
+import org.mapstruct.Mapper
+import org.mapstruct.Mapping
+import org.mapstruct.Mappings
+import org.mapstruct.Qualifier
+import org.mapstruct.ReportingPolicy
+import org.springframework.security.crypto.password.PasswordEncoder
+import org.springframework.stereotype.Component
+
+@Qualifier
+@Target(AnnotationTarget.FUNCTION)
+annotation class EncodedMapping
+
+@Component
+class PasswordEncoderMapper(private val passwordEncoder: PasswordEncoder) {
+  @EncodedMapping
+  fun encode(value: String): String {
+    return passwordEncoder.encode(value)
+  }
+}
+
+@Mapper(
+  componentModel = "spring",
+  uses = [PasswordEncoderMapper::class],
+  injectionStrategy = InjectionStrategy.CONSTRUCTOR,
+  unmappedTargetPolicy = ReportingPolicy.IGNORE
```

Comment:
I suggest using [ReportingPolicy.ERROR](https://mapstruct.org/documentation/stable/reference/html/#configuration-options). 
When you really want to ignore the mapping, you should explicitly ignore the field you want to ignore and every time the source is the same as the target, you just omit it.